org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js

```diff
@@ -0,0 +1,300 @@
+    */
+    putDeviceDetailsAsync(deviceId, putPayload) {
+
+        // build request path and body
+        var statusChanges = { device_id: deviceId };
+        var nonStatusChanges = {};
+        
+        for(var item in putPayload){
+            if(this._isDeviceStatus(item)){
+                statusChanges[item] = putPayload[item];
+            } else {
+                nonStatusChanges[item] = putPayload[item];
+            }
+        }
+
+        var translatorCopy = new Translator(this._accessToken);
+        var promises = [];
+
+        if (Object.keys(statusChanges).length > 0) {
+            var statePromise = this._makeRequest(this._commandPath, 'POST', JSON.stringify(statusChanges))
+                    .then(function (response) {
+                        return translatorCopy._getCommandResponse(response.id, 3) // get command status
+                            .then((commandResult) => {
+                                return Promise.resolve(commandResult);
+                            });
+                    });
+            promises.push(statePromise);
+        }
+
+        if (Object.keys(nonStatusChanges).length > 0) {
+            var nonStatePromise = this._makeRequest(this._devicesPath + '/' + deviceId, 'GET')
+                    .then((details) => {
+
+                        for (var item in nonStatusChanges) {
+                            details[item] = nonStatusChanges[item];
+                        }
+
+                        return this._makeRequest(this._devicesPath + '/' + deviceId, 'PUT', JSON.stringify(details))
+                                .then(function (response) {
+                                    return Promise.resolve(response);
+                                });
+                    });
+            promises.push(nonStatePromise);
+        }
+
+
+        // Merge Responses
+        var device = {};
+        return Promise.all(promises).then(function (responses) {
+            
+            for (var i = 0; i < responses.length ; i++) {
+                if ( responses[i] !== undefined && responses[i].status == 'succeeded') {
+                    if (responses[i].command !== undefined)
+                    {
+                        if( responses[i].command.level !== undefined) {
+                            device['Level'] = responses[i].command.level;
+                        }
+
+                        if (responses[i].command !== undefined && responses[i].command.command !== undefined) {
+                            device['Power'] = responses[i].command.command;
+                        }
+                    }
+
+                    if (responses[i].content !== undefined) {
+                        for (var item in responses[i].content) {
+                            device[item] = responses[i].content[item];
+                        }
+                    }
+                }
+            }
+            
+            if(device.length == 0 ) return Promise.resolve(undefined);
```

Comment:
Fixed.

---
In reply to: [82905445](https://github.com/openT2T/translators/pull/107#discussion_r82905445) [](ancestors = 82905445)